src/main/cli/cli.ts

```diff
@@ -122,48 +135,20 @@ export async function spawnSubcommand(
 
               cli.on('exit', async (code, signal) => {
                 logger.info({ subCommand, code, signal }, `CLI excited`);
-
+                onExit(code, signal);
+                resolveRunning();
                 cli = null;
-
-                try {
-                  await readFromDatabaseAndUpdateState();
-                } finally {
-                  onExit(code, signal);
-                  resolveRunning();
-                }
               });
 
               [cli.stderr, cli.stdout].forEach((stream) => {
                 stream.setEncoding('utf8');
                 stream.on('data', (data: string) => {
                   onStdOut(data);
 
-                  const logs = getLinesOfString(data)
-                    .map((logText) => {
-                      logger.debug(
-                        { subCommand, logText: logText.trim() },
-                        'Received stdout from cli process'
-                      );
-
-                      try {
-                        return JSON.parse(logText);
-                      } catch (err) {
-                        logger.debug(
-                          {
-                            subCommand,
-                            logText,
-                            err,
-                          },
-                          'Failed to parse CLI log'
-                        );
-                      }
-                      return null;
-                    })
-                    .filter(isCliLog);
+                  logger.debug({ subCommand, data }, `CLI stdout`);
+                  const logs = getLogsFromRawFileString(data);
 
                   onLog(logs);
-
-                  readFromDatabaseAndUpdateState();
                 });
               });
             } catch (e) {
```

Comment:
**Suggestion:** Consider handling the error in the catch block at line 154. Currently, the error is caught but not handled, which could lead to unexpected behavior.
```suggestion
try {
  ...
} catch (e) {
  logger.error(`Error occurred: ${e.message}`);
}
```